core/src/com/unciv/models/ruleset/unique/Conditionals.kt

```diff
@@ -188,8 +189,9 @@ object Conditionals {
             UniqueType.ConditionalWhenGarrisoned ->
                 checkOnCity { getCenterTile().militaryUnit?.canGarrison() == true }
 
-            UniqueType.ConditionalVsCity -> state.theirCombatant?.matchesFilter("City") == true
-            UniqueType.ConditionalVsUnits,  UniqueType.ConditionalVsCombatant -> state.theirCombatant?.matchesFilter(condition.params[0]) == true
+            UniqueType.ConditionalVsCity -> state.theirCombatant?.let {
+                it is CityCombatant && it.city.matchesFilter(condition.params[0]) } == true
+            UniqueType.ConditionalVsUnits,  UniqueType.ConditionalVsCiv -> state.theirCombatant?.matchesFilter(condition.params[0]) == true
```

Comment:
1. `VsUnits` should also have an `it is MapUnitCombatant` check
2. I don't really like having `vs civFilter` counting on combatant.matchesFilter. If anything, we should be using `combatant.getCivInfo.matchesFilter`

Reply:
I see, we can dispose of the ICombatant.matchesFilter entirely
Correct!